akka-stream-tests/src/test/scala/akka/stream/scaladsl/SubstreamSubscriptionTimeoutSpec.scala

```diff
@@ -74,7 +73,8 @@ class SubstreamSubscriptionTimeoutSpec(conf: String) extends AkkaSpec(conf) {
       // sleep long enough for it to be cleaned up
       Thread.sleep(1500)
 
-      val f = s3.runWith(Sink.head).recover { case _: SubscriptionTimeoutException ⇒ "expected" }
+      // Must be a Sink.seq, otherwise there is a race due to the concat in the `lift` implementation
+      val f = s3.runWith(Sink.seq).recover { case _: SubscriptionTimeoutException ⇒ "expected" }
```

Comment:
Well, I can see how this may prevent a certain race condition from being relevant, but there still is another, no? I.e. this test will not be fully deterministic if the scheduled timeout has not happened within the duration that this test procedure thread has slept (plus signal propagation delays).